app/Http/Controllers/Auth/AuthenticatedSessionController.php

```diff
@@ -50,14 +54,43 @@ public function store(Request $request)
         ]);
     }
 
+    /**
+     * Handles user authentication for socials
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function authenticateSocials(Request $request)
+    {
+        $user = User::where('email', $request->email)->first();
+
+        if (!$user) {
+            $user = User::create([
+                'id' => (string) Str::orderedUuid(),
+                'username' => $request->name,
+                'email' => $request->email,
+                'password' => Hash::make(Random::generate(20)),
+                'image' => $request->picture
+            ]);
+        }
+
+        $token = $user->createToken('token')->plainTextToken;
```

Comment:
This can be an inline variable in the json response: 'token' => $user->createToken('token')->plainTextToken